Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt

```diff
@@ -0,0 +1,152 @@
+    private val vm: HomeFragmentViewModel by cwaViewModels { viewModelFactory }
+
+    private val binding: FragmentHomeBinding by viewBindingLazy()
+
+    @Inject lateinit var homeMenu: HomeMenu
+    @Inject lateinit var tracingExplanationDialog: TracingExplanationDialog
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.tracingViewModel = vm.tracingViewModel
+        binding.settingsViewModel = vm.settingsViewModel
+        binding.submissionViewModel = vm.submissionViewModel
+
+        setupToolbar()
+
+        setupTestResultCard()
+
+        binding.mainTracing.setOnClickListener {
+            findNavController().doNavigate(HomeFragmentDirections.actionMainFragmentToSettingsTracingFragment())
+        }
+
+        setupRiskCard()
+
+        binding.mainAbout.mainCard.apply {
+            setOnClickListener {
+                ExternalActionHelper.openUrl(this@HomeFragment, getString(R.string.main_about_link))
+            }
+            contentDescription = getString(R.string.hint_external_webpage)
+        }
+
+        vm.showInteropDeltaOnboarding.observe2(this) {
+            findNavController().doNavigate(
+                HomeFragmentDirections.actionMainFragmentToOnboardingDeltaInteroperabilityFragment()
+            )
+        }
+
+        vm.showTracingExplanation.observe2(this) { activeTracingDaysInRetentionPeriod ->
+            tracingExplanationDialog.show(activeTracingDaysInRetentionPeriod) { vm.tracingExplanationWasShown() }
+        }
+
+        vm.showErrorResetDialog.observe2(this) {
+            RecoveryByResetDialogFactory(this).showDialog(
+                detailsLink = R.string.errors_generic_text_catastrophic_error_encryption_failure,
+                onDismiss = { vm.errorResetDialogDismissed() }
+            )
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // refresh required data
+        vm.refreshRequiredData()
+        binding.mainScrollview.sendAccessibilityEvent(AccessibilityEvent.TYPE_ANNOUNCEMENT)
+    }
+
+    private fun setupRiskCard() {
+        binding.mainRisk.apply {
+            riskCard.setOnClickListener {
+                findNavController().doNavigate(
+                    HomeFragmentDirections.actionMainFragmentToRiskDetailsFragment()
+                )
+            }
+            riskCardButtonUpdate.setOnClickListener {
+                vm.tracingViewModel.refreshDiagnosisKeys()
+                vm.settingsViewModel.updateManualKeyRetrievalEnabled(false)
+            }
+            riskCardButtonEnableTracing.setOnClickListener {
+                findNavController().doNavigate(
+                    HomeFragmentDirections.actionMainFragmentToSettingsTracingFragment()
+                )
+            }
+        }
+    }
+
+    private fun setupTestResultCard() {
+        binding.apply {
+            val toSubmissionResult = {
```

Comment:
val not fun ?

Reply:
The `val` value here is the anonymous function we are declaring. I find this more readable and light weight than using a `fun` function type, which requires more to keep the linter from complaining.

I'd only use the later part for more complex scenarios with nested returns.

![Screenshot from 2020-10-06 16-31-09](https://user-images.githubusercontent.com/1439229/95215699-9009b480-07f1-11eb-8a23-09b07c3651cc.png)

https://stackoverflow.com/questions/48112081/why-does-kotlin-have-two-syntaxes-for-lambdas-anonymous-functions